invenio_app_ils/records/api.py

```diff
@@ -144,6 +145,22 @@ def create(cls, data, id_=None, **kwargs):
         }
         return super(Item, cls).create(data, id_=id_, **kwargs)
 
+    def delete(self, **kwargs):
+        """Delete Item record."""
+        loan_search_res = search_by_pid(
+            item_pid=self[Item.pid_field],
+            filter_states=["ITEM_ON_LOAN"]
```

Comment:
It is probably safer to use `current_app.config['CIRCULATION_STATES_LOAN_ACTIVE']`